### PR TITLE
Use Mocha in Tests

### LINKED ISSUE #121: Feature - Switch to mocha for testing
**Is your feature request related to a problem? Please describe.**
EpiJS cannot find errors for NodeJS, only the web or errors that span both.

**Describe the solution you'd like**
 Switching to mocha allows us to use test NodeJS, and continue to test the web with puppeteer. 

**Describe alternatives you've considered**
Manually testing, though that does not give us the same level of guarantee that there are no bugs, and code coverage cannot be reported.

**Additional context**
I am working on this in the `v3-development` branch.

### PR DESCRIPTION
**Pull request by Quantalabs**
## Checklist:
- [x] Title is meaningful
- [x] Followed Contributing Guidelines
- [x] No _new_ issues, pre-existing issues in the repo are okay. We just don't want faulty code to be published to the nightly build or main if it can be prevented
- [x] All new dependencies are added to `[package.json]`
 
## Overview
Adds `mocha` and `istanbul` to testing, and up code coverage to 100%.

## Other Notes

Closes #121

### CODE REVIEW

File: EpiJS/comp.js [hunk 1]
```diff
@@ -61,6 +61,8 @@ class Idiom {
     let out = math.evaluate(parsed.join(''))
 
     // Round up to 0 if below 0 for out
+		/* istanbul ignore if */
+		/* istanbul ignore if */
```
Comment (quantum9Innovation) [inline EpiJS/comp.js:65]: Why is this being ignored?

File: EpiJS/comp.js [hunk 2]
```diff
@@ -639,7 +647,7 @@ class Dead {
  *      // Note that you can pass in a string as a rate too, 
  *      // but we use a number in this case because we don't need to multiply 
  *      // by other compartment populations. We do actually do this for the prev parameter, though.
- *      let I = new Infected([0.001], ["S*0.4"], false)
+ *      let V = new Vaccinated([0.001], ["S*0.4"], false)
```
Comment (quantum9Innovation) [inline EpiJS/comp.js:650]: How is this related to testing?

File: EpiJS/model.js [hunk 3]
```diff
@@ -212,16 +213,18 @@ function mexport(model, output, file_type=".json") {
...
  */
 function mimport(input, file_type=".json") {
+	let json = null;
 	if (file_type === ".json") {
-		let json = JSON.parse(fs.readFileSync(input))
+		json = JSON.parse(fs.readFileSync(input))
 	} else if (file_type === ".js") {
-		let json = require(output)
+		json = require(input)
 	}
```
Comment (quantum9Innovation) [inline EpiJS/model.js:221]: How has testing motivated this change?

File: EpiJS/model.js [hunk 4]
```diff
@@ -189,6 +189,7 @@ function mexport(model, output, file_type=".json") {
 		"key": model.key
 	}
 	for (let x in model.compartments) {
+		/* istanbul ignore if */
```
Comment (quantum9Innovation) [inline EpiJS/model.js:192]: Why is this being ignored?

File: EpiJS/plots.js [hunk 5]
```diff
@@ -71,7 +71,7 @@ function plot(model, time, name, title='Cases vs. Time') {
             mode: 'lines'
         })
     }
-
+    /* istanbul ignore if */
```
Comment (quantum9Innovation) [inline EpiJS/plots.js:74]: Again here ... and later in the same file

File: package.json [hunk 6]
```diff
@@ -27,8 +29,10 @@
     "homepage": "epi.js.org",
     "devDependencies": {
         "browserify": "~17.0.0",
+        "codecov": "^3.8.3",
         "documentation": "~13.2.5",
         "mocha": "~9.2.0",
+        "nyc": "^15.1.0",
```
Comment (quantum9Innovation) [inline package.json:35]: Why is `nyc` necessary here?

File: tests/index.js [hunk 7]
```diff
@@ -1,30 +1,261 @@
+const assert = require('assert');
+const epijs = require('..');
+const package = require('../package.json');
 const puppeteer = require('puppeteer');
+const { doesNotMatch } = require('assert');
```
Comment (quantum9Innovation) [inline tests/index.js:5]: Import `doesNotMatch` is never used
```suggestion

```

**Review by quantum9Innovation — COMMENTED** [review]
Many unnecessary changes (perhaps consider creating another PR?)

File: EpiJS/model.js (hunk 3 above)
Comment (Quantalabs) [reply EpiJS/model.js:221]: New bugs were found and fixed, this being one of them.

File: package.json (hunk 6 above)
Comment (Quantalabs) [reply package.json:35]: Nyc is the CLI interface for istanbul.

File: EpiJS/comp.js (hunk 2 above)
Comment (Quantalabs) [reply EpiJS/comp.js:650]: New bugs were found and fixed, this being one of them.

File: EpiJS/comp.js (hunk 1 above)
Comment (Quantalabs) [reply EpiJS/comp.js:65]: This cannot be assured to run every single time, so we ignore it.

File: EpiJS/model.js (hunk 4 above)
Comment (Quantalabs) [reply EpiJS/model.js:192]: This can't be assured to run every time we run a test, thus is ignored.

File: EpiJS/plots.js (hunk 5 above)
Comment (Quantalabs) [reply EpiJS/plots.js:74]: We can't test things running on the web.

**Review by quantum9Innovation — APPROVED** [review]
